upgrade_dependencies.py

```
@@ -0,0 +1,113 @@
+
+        print()
+
+    if dry_run:
+        print("Dry-run complete. Run without --dry-run to apply changes.")
+        print("After applying, run:")
+        print("  poetry lock")
+        print("  poetry install")
+        print("  poetry run pytest -q")
+    else:
+        print(f"Upgraded {upgraded_count} dependencies. Running poetry lock...")
+        subprocess.check_call(["poetry", "lock"])
```

Ruff (0.15.13)

[error] Starting a process with a partial executable path

(S607)